src/compiler/scala/tools/nsc/ast/parser/Parsers.scala

```diff
@@ -1255,9 +1264,12 @@ self =>
      */
     def literal(isNegated: Boolean = false, inPattern: Boolean = false, start: Offset = in.offset): Tree = atPos(start) {
       def finish(value: Any): Tree = try newLiteral(value) finally in.nextToken()
-      if (in.token == SYMBOLLIT)
-        Apply(scalaDot(nme.Symbol), List(finish(in.strVal)))
-      else if (in.token == INTERPOLATIONID)
+      if (in.token == SYMBOLLIT) {
+        if(settings.YliteralTypes)
+          finish(Symbol(in.strVal.intern()))
```

Comment:
I'd prefer to represent these with a distinct type from `Symbol` to avoid a memory leak in a resident compiler (`Symbol`-s are stored in a hash map).